frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -357,17 +358,60 @@ private void downloadLicenseInfo(ResourceRequest request, ResourceResponse respo
             final LicenseInfoService.Iface licenseInfoClient = thriftClients.makeLicenseInfoClient();
             final User user = UserCacheHolder.getUserFromRequest(request);
             Project project = thriftClients.makeProjectClient().getProjectById(projectId, user);
+            includeReleasesByChosenRelationship(request, releaseIdsToSelectedAttachmentIds);
             LicenseInfoFile licenseInfoFile = licenseInfoClient.getLicenseInfoFile(project, user, outputGenerator,
                     releaseIdsToSelectedAttachmentIds, excludedLicensesPerAttachmentId, externalIds);
             saveLicenseInfoAttachmentUsages(project, user, selectedAttachmentIdsWithPath,
-                    excludedLicensesPerAttachmentIdWithPath);
+                    excludedLicensesPerAttachmentIdWithPath, getSelectedReleaseRationships(request));
             sendLicenseInfoResponse(request, response, project, licenseInfoFile);
         } catch (TException e) {
             log.error("Error getting LicenseInfo file for project with id " + projectId + " and generator " + outputGenerator, e);
             response.setProperty(ResourceResponse.HTTP_STATUS_CODE, Integer.toString(HttpServletResponse.SC_INTERNAL_SERVER_ERROR));
         }
     }
 
+    private void includeReleasesByChosenRelationship(ResourceRequest request,
+            final Map<String, Set<String>> releaseIdsToSelectedAttachmentIds) {
+        final List<String> relationshipsToBeIncluded = getSelectedReleaseRationships(request).stream()
+                .map(rel -> ThriftEnumUtils.enumByString(rel, ReleaseRelationship.class)).filter(Objects::nonNull)
+                .map(ReleaseRelationship::name).collect(Collectors.toList());
+
+        Set<String> currentReleaseIdRelationships = Sets
+                .newHashSet(request.getParameterValues(PortalConstants.PROJECT_RELEASE_RELATIONS));
+
+        Map<String, String> currentReleaseIdToRelationship = extractCurrentReleaseIdToRelationship(
+                currentReleaseIdRelationships);
+
+        currentReleaseIdToRelationship = currentReleaseIdToRelationship.entrySet().stream()
+                .filter(x -> relationshipsToBeIncluded.contains(x.getValue()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        releaseIdsToSelectedAttachmentIds.keySet().retainAll(currentReleaseIdToRelationship.keySet());
+    }
+
+    private List<String> getSelectedReleaseRationships(ResourceRequest request) {
+        List<String> selectedReleaseRelationships = Lists.newArrayList();
+        Optional<String> relationshipsToBeIncluded = Optional
+                .ofNullable(request.getParameter(PortalConstants.SELECTED_PROJECT_RELEASE_RELATIONS));
+
+        if (relationshipsToBeIncluded.isPresent() && StringUtils.isNotEmpty(relationshipsToBeIncluded.get())) {
```

Comment:
Please remove Optional and use `CommonUtils.isNullEmptyOrWhitespace()`

Reply:
Done